lib/node_modules/@stdlib/repl/lib/multiline_handler.js

```diff
@@ -0,0 +1,600 @@
+	if ( this._lineIndex >= this._lines.length - 1 ) {
+		return;
+	}
+	this._cmd[ this._lineIndex ] = this._rli.line; // update current line in command
+
+	// Move cursor to the start of the next line:
+	this._moveCursor( 0, 1 );
+});
+
+/**
+* Simulates a backspace by removing the current line and appending it to the previous line.
+*
+* @private
+* @name _backspace
+* @memberof MultilineHandler.prototype
+* @type {Function}
+* @returns {void}
+*/
+setNonEnumerableReadOnly( MultilineHandler.prototype, '_backspace', function backspace() {
+	var cursor;
+
+	// If already at the first line, ignore...
+	if ( this._lineIndex <= 0 ) {
+		return;
+	}
+	// Save cursor position:
+	cursor = this._cmd[ this._lineIndex - 1 ].length;
+
+	// Append current line to the previous line:
+	this._cmd[ this._lineIndex - 1 ] += this._rli.line;
+	this._lines[ this._lineIndex - 1 ] += this._lines[ this._lineIndex ];
+
+	// Remove current line:
+	this._cmd.splice( this._lineIndex, 1 );
+	this._lines.splice( this._lineIndex, 1 );
+
+	// Move cursor to the saved cursor position in the previous line:
+	this._moveCursor( cursor, -1 );
+
+	// If we deleted all additional lines, update flag...
+	if ( this._lines.length <= 1 ) {
+		this._multiline.active = false;
+	}
+});
+
+/**
+* Resets input buffers.
+*
+* @private
+* @name _resetInput
+* @memberof MultilineHandler.prototype
+* @type {Function}
+* @returns {void}
+*/
+setNonEnumerableReadOnly( MultilineHandler.prototype, '_resetInput', function resetInput() {
+	this._cmd.length = 0;
+	this._lineIndex = 0;
+	this._lines.length = 0;
+});
+
+/**
+* Updates flags and buffers before to trigger multiline mode.
+*
+* @private
+* @name _triggerMultiline
+* @memberof MultilineHandler.prototype
+* @type {Function}
+* @returns {void}
+*/
+setNonEnumerableReadOnly( MultilineHandler.prototype, '_triggerMultiline', function triggerMultiline() {
+	// Update flag:
+	this._multiline.trigger = true;
+
+	// Save expression after cursor in buffer:
+	readline.clearLine( this._ostream, 1 ); // clear line after cursor
+	this._remainingLine = this._rli.line.substring( this._rli.cursor );
+	this._rli.line = this._rli.line.substring( 0, this._rli.cursor );
+});
+
+/**
+* Checks if the command is incomplete and a multiline input.
+*
+* @private
+* @name _isMultilineInput
+* @memberof MultilineHandler.prototype
+* @type {Function}
+* @param {string} cmd - command
+* @returns {boolean} boolean indicating whether the command is a multiline input
+*/
+setNonEnumerableReadOnly( MultilineHandler.prototype, '_isMultilineInput', function isMultilineInput( cmd ) {
+	var node;
+	var tmp;
+	var ast;
+
+	debug( 'Attempting to detect multi-line input...' );
+	if ( RE_WHITESPACE.test( cmd ) ) {
+		debug( 'Detected multi-line input. Triggering multiline mode...' );
+		return true;
+	}
+	if ( RE_SINGLE_LINE_COMMENT.test( cmd ) || RE_MULTI_LINE_COMMENT.test( cmd ) ) { // eslint-disable-line max-len
+		debug( 'Multi-line input not detected.' );
+		return false;
+	}
```

Comment:
@kgryte I was also thinking why were we not treating comments as a multiline input? Does it not make sense to allow more lines if a comment is encountered?